### ISSUE TITLE
pgloader 3.6.1 compiling error with sbcl stable 2.0.1

### ISSUE THREAD

**Issue by tigerfoot** [dimitri/pgloader#1087]
- [X] pgloader --version
    ```
3.6.1 (bundle source)
    ```
  
  - [X] did you test a fresh compile from the source tree?
For our openSUSE Tumbleweed rolling release, every package is build every time on a new and fresh system (ala travis)
    
  - [X] did you search for other similar issues?
Have search sbcl 2 issues here but not found

  - [X] how can I reproduce the bug?
Try to compile pgloader with the last stable sbcl which is now at version 2.0.1 

You can have a look at the whole build (and failure) at the following
[Build log on obs](https://build.opensuse.org/build/home:bruno_friedmann:branches:server:database:postgresql/openSUSE_Tumbleweed/x86_64/pgloader/_log)

**warnings** about depreciation, that one day will become error, and we need to fix ;-)

```
[   58s] WARNING:
[   58s]    Lisp compilation had style-warnings while
[   58s]    compiling #<CL-SOURCE-FILE "buildapp" "buildapp">
[   58s] WARNING:
[   58s]    Reference to deprecated function (SB-DEBUG:BACKTRACE-AS-LIST) from COMMAND-LINE-DEBUGGER
```

**fatal error**

```
[   58s] ;; loading system "buildapp"
[   63s] bin/buildapp      --logfile /tmp/pgloader-bundle-build.log     \
[   63s]                          --require sb-posix --require sb-bsd-sockets --require sb-rotate-byte                             \
[   63s]                          --sbcl sbcl                                 \
[   63s]                          --asdf-tree .                                \
[   63s]                          --load-system cffi                           \
[   63s]                          --load-system cl+ssl                         \
[   63s]                          --load-system mssql                          \
[   63s]                          --load local-projects/pgloader-3.6.1/src/hooks.lisp              \
[   63s]                          --load-system pgloader                    \
[   63s]                          --eval '(setf pgloader.params::*version-string* "3.6.1")' \
[   63s]                          --entry pgloader:main                        \
[   63s]                          --dynamic-space-size 4096              \
[   63s]                          --compress-core                         \
[   63s]                          --output bin/pgloader.tmp
[   63s] ;; loading system "cffi"
[   72s] Fatal SIMPLE-ERROR:
[   72s]   Compilation failed: :LINKAGE-TABLE is no longer present in *FEATURES* in /home/abuild/rpmbuild/BUILD/pgloader-bundle-3.6.1/software/cffi_0.20.0/src/cffi-sbcl.lisp
[   72s] make: *** [Makefile:47: bin/pgloader] Error 1
[   72s] error: Bad exit status from /var/tmp/rpm-tmp.Qn6Vgx (%build)
[   72s] 
```

I've no real knowledge about lisp so any pointer or patches that can be applied would be great.
I guess this type of fix + pg12 support will drive us to a new version?

**Comment by tigerfoot** [comment]
Any news ? Unfortunately, if I don't fix the build compile error on Tumbleweed soon, the package will have to be removed, I would be really worried about that.

**Comment by tigerfoot** [comment]
Like for example release a new version with cffi version 0.21
https://github.com/cffi/cffi/releases/tag/v0.21.0
```SBCL no longer pushes :linkage-table to *features*```

**Comment by phoe** [comment]
Not related to pgloader itself. 

This is an error related to SBCL removing the `*LINKAGE-TABLE*` feature in its newest versions.

Please update CFFI to a new version, preferably one that includes the commit https://github.com/cffi/cffi/commit/375872a86001f970156f8390ff5ef80df01901f5 that fixes this issue.

**Comment by tigerfoot** [comment]
cffi is contained in the pgloader bundle available as release, so still a bit related to pgloader ;-)
That's why (plus all the issues closed since last 3.6.1) I've feel that the time to a new release is coming.

**Comment by phoe** [comment]
A workaround would be to use an older SBCL version, e.g. 2.0.0 or earlier.

**Comment by tigerfoot** [comment]
Unfortunately I don't know a packager able to downgrade a software version on a mainstream distribution, sbcl maintainer at openSUSE decide that 2.0.1 is the latest stable (and upstream too), so do I accept and live with ;-) 
I've just make a test with cffi 0.21.0 replaced in the bundle and now I get pgloader building again, so a temporary fix.

**Comment by phoe** [comment]
I'll track the new release at #1092.

**Comment by tigerfoot** [comment]
Thanks I'm closing this one so.

**Comment by phoe** [comment]
I'll allow myself to reopen, since we only have a workaround and not a proper fix.

**Comment by phoe** [comment]
Fixed in #1092. Please use the freshly released version 3.6.2 that is buildable with the newer SBCL versions.

**Comment by tigerfoot** [comment]
Super, big thanks for the work.
